engine/proxy.go

```diff
@@ -0,0 +1,227 @@
+package engine
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strconv"
+	"time"
+
+	"github.com/prisma/prisma-client-go/binaries"
+	"github.com/prisma/prisma-client-go/logger"
+	"github.com/prisma/prisma-client-go/runtime/types"
+)
+
+func NewDataProxyEngine(schema, schemaEnvVarName string) *DataProxyEngine {
+	return &DataProxyEngine{
+		Schema:           schema,
+		schemaEnvVarName: schemaEnvVarName,
+		http:             &http.Client{},
+	}
+}
+
+type DataProxyEngine struct {
+	// http is the internal http client
+	http *http.Client
+
+	// url holds the query-engine url
+	url string
+
+	// schemaEnvVarName is the env var for the datasource url
+	// this is needed internally to extract the api key
+	schemaEnvVarName string
+
+	// Schema contains the prisma Schema
+	Schema string
+
+	// apiKey contains the parsed prisma data proxy api key from the connection string
+	apiKey string
+}
+
+func (e *DataProxyEngine) Connect() error {
+	// Example uri: https://aws-eu-west-1.prisma-data.com/2.26.0/412bf0a1742a576d699fbd5102a4f725557eff3992995f2e18febce128794961/
+	hash := hashSchema(e.Schema)
+	logger.Debug.Printf("local schema hash %s", hash)
+
+	logger.Debug.Printf("parsing connectino string from database url %s", e.schemaEnvVarName)
```

Comment:
s/connectino/connection/
As much as I love `connectino`: it sounds like a small, Italian `connection`